### PR TITLE
Add lesson 12

### PR DESCRIPTION
**Pull request by karinanartowska**

### CODE REVIEW

**Comment by karinanartowska** [comment]
@gradopaw please review :)

File: students/nartowska_karina/lesson_12_debugging/verbose_output.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                        default='debug')
+
+    results = parser.parse_args(args)
+    return (results.regex,
+            results.verbose)
+
+
+regex, verbose = check_arg(sys.argv)
+logging.basicConfig(level=LOGGER_LEVELS[verbose],
+                    format='%(asctime)s - %(levelname)s - %(message)s')
```
Comment (gradopaw) [inline students/nartowska_karina/lesson_12_debugging/verbose_output.py]: There is no need to call twice ```check_arg(args=None):``` function in your program. You can move lines 29 - 31 to 48 line, in a result your ```main()``` function will be looks following:
```
def main():
    regex, verbose = check_arg(sys.argv[1:])
    logging.basicConfig(level=LOGGER_LEVELS[verbose],
                        format='%(asctime)s - %(levelname)s - %(message)s')
    logging.debug("Searching regex in .txt files: " + regex)
    regex_search = re.compile(regex)
    search_regex(regex_search)
```

File: students/nartowska_karina/lesson_12_debugging/validated_user_base.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+
+def write_to_csv(input):
+    with open('data.csv', 'a+') as csv_file:
+        fieldnames = ['email', 'password', 'phone_number', 'postal_code']
+        writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+
+        if not os.stat(csv_file.name).st_size:
+            writer.writeheader()
+        writer.writerow(input)
```
Comment (gradopaw) [inline students/nartowska_karina/lesson_12_debugging/validated_user_base.py]: According to the task description, there shouldn't be possibility to insert data with the same address more than one time:

If a record with the same email is already in the file, the old record should be altered by new one

File: students/nartowska_karina/lesson_12_debugging/verbose_output.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        for line in content:
+            if regex_search.match(line):
+                logging.debug("Correct line is:\n" +
+                              line + "from file: " + file)
+
+        current.close()
+
+
+def main():
+    reg = check_arg(sys.argv[1:])
```
Comment (gradopaw) [inline students/nartowska_karina/lesson_12_debugging/verbose_output.py]: In this case, reg is a tuple so next line (line 50 as well) causing an error. 
```reg = check_arg(sys.argv[1:])```

**Comment by karinanartowska** [comment]
@gradopaw please review :)

**Pull request merged**